File: Formations/Boucles/Random Loops/Solutions/4/1.php
<?php
$cible = rand(0, 100);
$proposition = -1;
$essais = 0;

while ($proposition !== $cible) {
    $proposition = readline("Proposez une valeur : ");
    $essais++;

    if ($proposition < $cible) {
        echo "Trop petit.\n";
    } elseif ($proposition > $cible) {
        echo "Trop grand.\n";
    } else {
        echo "Bravo, vous avez trouvé le nombre en $essais coup(s).\n";
    }
}

echo "En $essais coups, vous êtes sûr(e) de trouver.\n";
?>

File: Formations/Boucles/Random Loops/Solutions/4/1.py
import random

nombre_cible = random.randint(0, 100)
proposition_utilisateur = -1
essais = 0

while proposition_utilisateur != nombre_cible:
    proposition_utilisateur = int(input("Proposez une valeur : "))
    essais += 1

    if proposition_utilisateur < nombre_cible:
        print("Trop petit.")
    elif proposition_utilisateur > nombre_cible:
        print("Trop grand.")
    else:
        print("Bravo, vous avez trouvé le nombre en", essais, "coup(s).")

print("En", essais, "coups, vous êtes sûr(e) de trouver.")
